pcdsdevices/attenuator.py

```diff
@@ -1104,6 +1139,170 @@ def format_status_info(self, status_info):
+    This class also includes control for the Kurt J. Lesker LBD stage.
+
+    Parameters
+    ----------
+    prefix : str
+        Solid Attenuator base PV.
+
+    name : str
+        Alias for the Solid Attenuator.
+
+    calculator_prefix : str
+        The prefix for the calculator PVs.
+    """
+
+    # QIcon for UX
+    _icon = 'fa.barcode'
+    tab_component_names = True
+
+    # Register that all blades are needed for lightpath calc
+    lightpath_cpts = [f'blade_{idx:02}.state.state' for idx in range(1, 4)]
+
+    # Summary for lightpath view
+    num_in = Cpt(InternalSignal, kind='hinted')
+    num_out = Cpt(InternalSignal, kind='hinted')
+
+    calculator = UCpt(AttenuatorCalculatorSXR_TwoBlade)
+    blade_01 = Cpt(SXRLadderAttenuatorBlade, ':MMS:01')
+    blade_02 = Cpt(SXRLadderAttenuatorBlade, ':MMS:02')
+    # LBD Stage
+    blade_03 = Cpt(FEESolidAttenuatorBlade, ':MMS:03')
+
+    def __init__(self, *args, limits=None, **kwargs):
+        UCpt.collect_prefixes(self, kwargs)
+        limits = limits or (0.0, 1.0)
+        super().__init__(*args, limits=limits, **kwargs)
+
+    @property
+    def setpoint(self):
+        """(PVPositioner compat) - use desired transmission as setpoint."""
+        return self.calculator.desired_transmission
+
+    @property
+    def readback(self):
+        """(PVPositioner compat) - use actual transmission as readback."""
+        return self.calculator.actual_transmission
+
+    @property
+    def actuate(self):
+        """(PVPositioner compat) - use apply_config as an actuation signal."""
+        return self.calculator.apply_config
+
+    def _setup_move(self, position):
+        """(PVPositioner compat) - calculate, then move."""
+        # Do not call `calculator.calculate()` here to respect the current
+        # calculator settings:
+        self.calculator.desired_transmission.put(position)
+        self.calculator.run_calculation.put(1, wait=True)
+        return super()._setup_move(position)
+
+    def get_lightpath_state(self, use_cache=True) -> LightpathState:
+        """
+        Grab slightly different PV values for use in same inout calc fn
+        The state is nested one device deeper than LightpathInOutCptMixin
+        expects.
+        """
+        if (not use_cache) or (self._cached_state is None):
+            lightpath_kwargs = {}
+            lp_sigs = self.lightpath_summary._signals.keys()
+            for sig in lp_sigs:
+                # want to get name of blade_0x from dev_blade_0x_state_state
+                cpt_name = sig.name.removeprefix(self.name + '_')
+                cpt_name = cpt_name.removesuffix('_state_state')
+                lightpath_kwargs[cpt_name] = sig.get()
+
+            self._cached_state = self.calc_lightpath_state(**lightpath_kwargs)
+
+        return self._cached_state
+
+    def calc_lightpath_state(self, **lightpath_kwargs) -> LightpathState:
+        # Repeat lightpath logic to extract num_in, num_out
```

Comment:
The logic here needs a bit of work as we've been discussing. Something along the lines of "use calculator transmission value if blade 3 is removed or 0.0 if blade 3 is inserted" - meaning that the number of lightpath components drops to 2 (calculator transmission and blade 3 state)

(Hmm, it could also be OK to merge this and then make a follow-up PR since this one is rather large...)

Reply:
Yeah it could be OK to handle this later alongside #1092 which needs much the same adjustment. It's possible that others also need a small tweak here.